src/main/scala/com/exini/dicom/data/Compression.scala

```diff
@@ -0,0 +1,43 @@
+package com.exini.dicom.data
+
+import akka.util.ByteString
+
+import java.util.zip.{ Deflater, Inflater }
+
+object Compression {
+
+  def compress(bytes: ByteString, compressor: Deflater): ByteString = {
+    compressor.setInput(bytes.toArray)
+    val buffer = new Array[Byte](bytes.length)
+    var out    = ByteString.empty
+    var n      = 1
+    while (n > 0) {
+      n = compressor.deflate(buffer, 0, buffer.length, Deflater.FULL_FLUSH)
+      out = out ++ ByteString.fromArray(buffer, 0, n)
+    }
+    out
+  }
+
+  def compress(bytes: ByteString, gzip: Boolean = false): ByteString = {
+    val compressor = if (gzip) new Deflater() else new Deflater(-1, true)
+    compress(bytes, compressor)
+  }
+
+  def decompress(bytes: ByteString, decompressor: Inflater): ByteString = {
+    decompressor.setInput(bytes.toArrayUnsafe())
```

Comment:
It's a good question. I've been looking carefully how inflation is done on the akka side. The corresponding line is https://github.com/akka/akka/blob/main/akka-stream/src/main/scala/akka/stream/impl/io/compression/DeflateDecompressorBase.scala#L35. Documentation of the method is here: https://github.com/akka/akka/blob/main/akka-stream/src/main/scala/akka/stream/impl/io/compression/DeflateDecompressorBase.scala#L35. It indicates that the "unsafe" part here is that when the ByteString is compact you get access to a direct copy of the underlying byte array so you have to take care not to modify this which would break immutability. But we are not, so we are saving some object allocation time and heap here.